imgui.cpp

```diff
@@ -2667,9 +2664,11 @@ static void ImGuiListClipper_SeekCursorForItem(ImGuiListClipper* clipper, int it
     ImGuiListClipper_SeekCursorAndSetupPrevLine(pos_y, clipper->ItemsHeight);
 }
 
-ImGuiListClipper::ImGuiListClipper()
+ImGuiListClipper::ImGuiListClipper(ImGuiContext* ctx)
 {
```

Comment:
Even consider further commit it seems reasonable for this commit to make this default to NULL and during construct turn NULL into current context?  Would prevent changing the demo code as well. 

ImGuiListClipper is a rare case of being instanciated on user-side so that could be reworked as well (would be nice to have an ImGui:: api for it that returned a struct on each step).

Reply:
@ocornut 

> Even consider further commit it seems reasonable for this commit to make this default to NULL and during construct turn NULL into current context? Would prevent changing the demo code as well.

Yes I agree, it is better better not to introduce any breaking change. Especially given the fact the breaking change is solved in the last commit.

The ideal way to avoid the breaking change and simplify the refactoring would be to introduce the implicit context free version of `ImGuiListClipper` and `ImGuiTextFilter` (i.e `ImGuiListClipperEx` and `ImGuiTextFilterEx`) as part of the first commits. By doing so, no work is required later. It is just a matter of moving the definition of surrounding ImGuiListClipper and ImGuiTextFilter with `#ifndef IMGUI_DISABLE_IMPLICIT_API`

Do you think it is acceptable?